engine/src/main/java/com/datatorrent/stram/plan/physical/StreamMapping.java

```diff
@@ -188,6 +192,69 @@ private void addSlidingUnifiers()
     }
   }
 
+  private List<PTOutput> setupPartitionerUnifiers(List<PTOutput> upstream, Partitioner<Operator> partitioner)
+  {
+    Operator.Unifier<?> unifier = streamMeta.getSource().getUnifier();
+
+    if (unifier == null) {
+      unifier = new DefaultUnifier();
+    }
+
+    Collection<Partitioner.Partition<Operator>> collection = new ArrayList<>(1);
+    DefaultPartition<Operator> firstPartition = new DefaultPartition<Operator>(unifier);
+    collection.add(firstPartition);
+
+    Collection<Partitioner.Partition<Operator>> partitions = partitioner.definePartitions(collection, new PhysicalPlan.UnifierPartitioningContextImpl());
+
+    if (partitions.size() <= 1) {
+      return upstream;
+    }
+    OperatorMeta om = streamMeta.getSource().getOperatorMeta();
+
+    InputPortMeta portMeta = null;
+    Map<InputPortMeta, StreamMeta> inputs = om.getInputStreams();
+    if (!inputs.isEmpty()) {
```

Comment:
Is the intent here to assign previously assigned keys if the previous component (connected by the stream) already has an assigned partitioning scheme? Can this be clarified with a comment?